bolero/environment/optimum_trajectory.py

```diff
@@ -294,5 +294,116 @@ def plot(self, ax):
         if self.obstacles is not None:
             from matplotlib.patches import Circle
             for obstacle in self.obstacles:
-                ax.add_patch(Circle(obstacle, self.obstacle_dist, ec="none",
+                ax.add_patch(Circle(np.asarray(obstacle).copy(), self.obstacle_dist, ec="none",
```

Comment:
Why is that required?

Reply:
This is necessary, when obstacle positions are changed but the user wants to control when the plot is updated.

I'll request merging this class later, which allows adapting the task accordingly:
class ContextualParetoOptimalTrajectory(OptimumTrajectory, ContextualEnvironment):
    def get_num_context_dims(self):
        return self.obstacles.size

    def get_context(self):
        return self.obstacles.ravel()

    def request_context(self, obstacles):
        if obstacles is None:
            return self.get_context()
        self.obstacles = np.reshape(obstacles, (-1, len(self.x0)))

Reply:
OK, but could you replace it by `np.copy(obstacle)`? I think `np.asarray(obstacle).copy()` copies obstacle twice if obstacle is not a numpy array.

Reply:
OK